src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
We may want to enhance this class to "control" which operations are allowed. Imagine:

```
using LedgerIndex = tagged_integer<std::uint32_t, ops::addition, ops::subtraction>;
using Number = tagged_integer<std::int32_t, ops::addition, ops::subtraction, ops::division, ops::modulo>;
```